contracts/marketplace/CruzoMarket.sol

```diff
@@ -48,6 +53,14 @@ contract CruzoMarket is ERC1155Holder, Ownable {
             itemToken.balanceOf(msg.sender, _itemId) != 0,
             "Error: Only owner can list"
         );
+
+        /*for (uint256 index = 0; index < addrToHisTrades[msg.sender].length; index++) {
+            require(addrToHisTrades[msg.sender][index] != _itemId);
+        }
+        addrToHisTrades[msg.sender].push(_itemId);
+        */
+        require(addrToItem[msg.sender][_itemId] == false, "already in trades");
+        addrToItem[msg.sender][_itemId] = true;
```

Comment:
Would it make sense to store the trades itself inside a nested mappings structure? Then we can get rid of the additional `addrToItem` mapping.